### ISSUE TITLE
TF 2.1/tf.keras AdaDelta optimizer: default epsilon and learning rate values

### ISSUE THREAD

**Issue by 8bitmp3** [tensorflow/tensorflow#36785]
Hi @tanzhenyu @lamberta @dynamicwebpaige There are two param default values for AdaDelta that look different from the original implementation. It is probably explained in past discussions (and if it is the case pls let me know where I can find it so we can close this issue.) Cheers.

Similar issues: https://github.com/tensorflow/tensorflow/issues/31024 and https://github.com/tensorflow/tensorflow/pull/31025.

TF 2.1 [API docs for the Adadelta optimizer](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adadelta) come from [`/optimizer-v2/`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/adadelta.py), where: 
- The learning rate is set to 0.001:
  - Keras' implementation states that learning rate is 1.
  - [`optimizer.py`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizers.py) (v1) sets it to 1 (synced with Keras, I suppose).
  - There's a Keras PR - https://github.com/keras-team/keras/pull/12841 - that proposed to change the LR from 1 to 0.001. It may have been reverted back to 1 (see commit https://github.com/keras-team/keras/pull/12888/commits/2009cab5217a57bfbb4dae88371640ce1bb4a0e9).
- The default epsilon does not match the one in the original AdaDelta [paper](https://arxiv.org/pdf/1212.5701.pdf) as well as Keras' implementation:
  - They use `1e-6` instead of `1e-7` (`"Setting the hyperparameters to ε = 1e − 6 and ρ = 0.95..."`).

Sources: 

- [TF 2.1 /optimizer_v2/](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/adadelta.py):

```python
@keras_export('keras.optimizers.Adadelta')
class Adadelta(optimizer_v2.OptimizerV2):
  ...
  def __init__(self,
               learning_rate=0.001,
               rho=0.95,
               epsilon=1e-7,
               name='Adadelta',
               **kwargs):
```

- [TF 2.1 optimizers.py - version 1](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizers.py):

```python
class Adadelta(Optimizer):
...
  def __init__(self, lr=1.0, rho=0.95, epsilon=None, decay=0., **kwargs):
...
```

- [Keras optimizers.py](https://github.com/keras-team/keras/blob/master/keras/optimizers.py):
```python
class Adadelta(Optimizer):
...
    def __init__(self, learning_rate=1.0, rho=0.95, **kwargs):
...
```

Thanks for taking your time to look at this potential issue.

**Comment by abhilash1910** [comment]
Hi @8bitmp3 , I also had a similar doubt when I came across the paper and the API documentation. I think according to the paper,the step size converges to 1 at end of training and the convergence occurs when the gradients and parameters updates are smaller. However there is a mismatch in the documentation which suggests that lr=0.001 whereas in the paper it is lr=1.0 and epsilon=1e-6. The less the LR I think it would be locally stuck in minima and not contribute significantly .

**Comment by abhilash1910** [comment]
Hi @8bitmp3 ,@gadagashwini, Have made a comment in the  https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/adadelta.py 
The comment states the original details as mentioned in the paper with LR 1 and epsilon 1e-6.
 PR #36832

**Comment by jvishnuvardhan** [comment]
@8bitmp3 I checked that @abhilash1910 PR https://github.com/tensorflow/tensorflow/pull/36849 got merged and I also reviewed that `AddaDelta` in `keras-team/keras` and `tf.keras` are in sync.

[Here](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adadelta?version=nightly) is the TF page on `AdaDelta` with the update.

[Here](https://github.com/keras-team/keras/blob/master/keras/optimizer_v2/adadelta.py) is the link to `AddaDelta` in `keras-team/keras`.
```
  def __init__(self,
               learning_rate=0.001,
               rho=0.95,
               epsilon=1e-7,
               name='Adadelta',
               **kwargs):
```
Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by 8bitmp3** [comment]
Thank you @jvishnuvardhan @lamberta 

`learning_rate=0.001`, `epsilon=1e-7`

👍 Closing the issue
